android/app/src/main/java/com/microsoft/reactnativecodepush/CodePush.java

```diff
@@ -0,0 +1,412 @@
+        }
+
+        try {
+            codePushPackage.rollbackPackage();
+        } catch (IOException e) {
+            throw new CodePushUnknownException("Error in rolling back package", e);
+        }
+
+        codePushNativeModule.loadBundle();
+    }
+
+    private void saveFailedUpdate(String packageHash) {
+        SharedPreferences settings = applicationContext.getSharedPreferences(CODE_PUSH_PREFERENCES, 0);
+        String failedUpdatesString = settings.getString(FAILED_UPDATES_KEY, null);
+        JSONObject failedUpdates;
+        if (failedUpdatesString == null) {
+            failedUpdates = new JSONObject();
+        } else {
+            try {
+                failedUpdates = new JSONObject(failedUpdatesString);
+            } catch (JSONException e) {
+                // Should not happen.
+                throw new CodePushMalformedDataException("Unable to parse failed updates information " +
+                        failedUpdatesString + " stored in SharedPreferences", e);
+            }
+        }
+        try {
+            failedUpdates.put(packageHash, true);
+            settings.edit().putString(FAILED_UPDATES_KEY, failedUpdates.toString()).commit();
+        } catch (JSONException e) {
+            // Should not happen unless the packageHash is null.
+            throw new CodePushUnknownException("Unable to save package hash " +
+                    packageHash + " as a failed update", e);
+        }
+    }
+
+    private void savePendingUpdate(String packageHash, int rollbackTimeout) {
+        SharedPreferences settings = applicationContext.getSharedPreferences(CODE_PUSH_PREFERENCES, 0);
+        JSONObject pendingUpdate = new JSONObject();
+        try {
+            pendingUpdate.put(PENDING_UPDATE_HASH_KEY, packageHash);
+            pendingUpdate.put(PENDING_UPDATE_ROLLBACK_TIMEOUT_KEY, rollbackTimeout);
+            settings.edit().putString(PENDING_UPDATE_KEY, pendingUpdate.toString()).commit();
+        } catch (JSONException e) {
+            // Should not happen.
+            throw new CodePushUnknownException("Unable to save pending update.", e);
+        }
+
+    }
+
+    private void startRollbackTimer(int rollbackTimeout) {
+        timer = new Timer();
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date());
+        c.add(Calendar.MILLISECOND, rollbackTimeout);
+        Date timeout = c.getTime();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                rollbackPackage();
+            }
+        }, timeout);
+    }
+
+    private class CodePushNativeModule extends ReactContextBaseJavaModule {
+
+        private void loadBundle() {
+            String assetsBundleFileUrl = CodePush.this.getBundleUrl(CodePush.this.assetsBundleFileName);
+            Intent intent = mainActivity.getIntent();
+            mainActivity.finish();
+            mainActivity.startActivity(intent);
```

Comment:
Not that I know of. Let me know if you found a way to do it. The assetsBundleFileUrl was leftover when I tried to find a way to do just that - invoke the bridge to navigate to the script. I will remove it.